In [7]:
import os
os.listdir("./archive")

['Training', 'Validation']

In [8]:
len(os.listdir("./archive/Training/male")), len(os.listdir("./archive/Training/female"))

(23766, 23243)

In [9]:
len(os.listdir("./archive/Validation/male")), len(os.listdir("./archive/Validation/female"))

(5808, 5841)

In [10]:
import torch.nn as nn
import torch

In [18]:
loss = nn.CrossEntropyLoss()
input = torch.randn(3, 2, requires_grad=True)
target = torch.empty(3, dtype=torch.long).random_(2)
input, target

(tensor([[ 0.7803,  1.2986],
         [-1.2384, -1.6242],
         [ 1.9469, -0.5054]], requires_grad=True),
 tensor([1, 1, 0]))

In [19]:
loss(input, target)

tensor(0.4848, grad_fn=<NllLossBackward0>)

(tensor([[-0.3466,  0.0829],
         [ 0.6925,  1.5139],
         [ 0.2762, -0.5167]], requires_grad=True),
 tensor([1, 0, 0]))
tensor(0.6868, grad_fn=<NllLossBackward0>)

(tensor([[-0.2174,  1.5132],
         [ 0.0669, -0.0062],
         [-1.2145, -0.7034]], requires_grad=True),
 tensor([0, 1, 1]))
 tensor(1.0313, grad_fn=<NllLossBackward0>)
 
(tensor([[ 0.7803,  1.2986],
         [-1.2384, -1.6242],
         [ 1.9469, -0.5054]], requires_grad=True),
 tensor([1, 1, 0]))
 tensor(0.4848, grad_fn=<NllLossBackward0>)